src/tests/catchy/test_base64.cpp

```diff
@@ -0,0 +1,67 @@
+#include "catch.hpp"
+#include <botan/base64.h>
+
+namespace {
+std::vector<Botan::byte> toStdVector(const Botan::secure_vector<Botan::byte> &in)
+   {
+   return std::vector<Botan::byte>(in.cbegin(), in.cend());
+   }
+
+std::vector<Botan::byte> toStdVector(const std::string &in)
+   {
+   return std::vector<Botan::byte>(in.cbegin(), in.cend());
+   }
+}
+
+TEST_CASE("Base64 encode empty string", "[base64]")
+   {
+   // common knowledge
+   auto emptyString = std::string("");
+   auto emptyVector = std::vector<Botan::byte>(emptyString.cbegin(), emptyString.cend());
+   REQUIRE(( Botan::base64_encode(emptyVector) == "" ));
+   }
+
+TEST_CASE("Base64 encode short string", "[base64]")
+   {
+   // test vectors from http://tools.ietf.org/html/rfc4648
+   auto in1 = std::vector<Botan::byte>{ 'f' };
+   auto in2 = std::vector<Botan::byte>{ 'f', 'o' };
+   auto in3 = std::vector<Botan::byte>{ 'f', 'o', 'o' };
+   REQUIRE(( Botan::base64_encode(in1) == "Zg==" ));
+   REQUIRE(( Botan::base64_encode(in2) == "Zm8=" ));
+   REQUIRE(( Botan::base64_encode(in3) == "Zm9v" ));
```

Comment:
I wonder if independent tests like this should be using `CHECK` assertion, with `REQUIRE` reserved for a test failure which obviates any further possible testing. For example if both tests 2 and 3 failed, that is useful information which would be lost by the `REQUIRE` on test 2 aborting the rest of the test case.